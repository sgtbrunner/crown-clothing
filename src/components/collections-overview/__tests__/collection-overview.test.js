import React from 'react';
import { Provider } from 'react-redux';
import { HashRouter as Router } from 'react-router-dom';
import { screen, render } from '@testing-library/react';
import configureMockStore from 'redux-mock-store';

import CollectionOverview from '../collections-overview.component';

const mockStore = configureMockStore();

const collections = [
  {
    id: '1',
    routeName: 'jackets',
    title: 'Jackets',
    items: [
      {
        id: 1,
        price: 10,
        name: 'Jacket 1',
        imageUrl: 'https://www.myimage.com/jacket1',
      },
      {
        id: 2,
        price: 20,
        name: 'Jacket 2',
        imageUrl: 'https://www.myimage.com/jacket2',
      },
    ],
  },
  {
    id: '2',
    routeName: 'shoes',
    title: 'Shoes',
    items: [
      {
        id: 1,
        price: 100,
        name: 'Shoes 1',
        imageUrl: 'https://www.myimage.com/shoes1',
      },
    ],
  },
];

const store = mockStore({
  shop: {
    collections,
  },
});

const mountComponent = () =>
  render(
    <Provider store={store}>
      <Router basename="/">
        <CollectionOverview />
      </Router>
    </Provider>
  );

describe('CollectionOverview component', () => {
  it('should match component snapshot', () => {
    const { container } = mountComponent();

    expect(container).toMatchSnapshot();
  });

  it('should render 2 (two) collection previews', () => {
    mountComponent();

    const previews = screen.getAllByTestId('collection-preview');
    expect(previews).toHaveLength(2);
  });

  it('should match rendered collection previews', () => {
    mountComponent();

    collections.forEach((collection) => {
      const collectionTitle = screen.getByRole('link', {
        name: collection.title.toUpperCase(),
      });
      expect(collectionTitle).toBeInTheDocument();
    });
  });

  it('should render 3 (three) collection items in total', () => {
    mountComponent();

    const items = screen.getAllByTestId('collection-item');
    expect(items).toHaveLength(3);
  });
});
