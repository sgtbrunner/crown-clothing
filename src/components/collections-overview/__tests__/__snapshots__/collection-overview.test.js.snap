// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CollectionOverview component should match component snapshot 1`] = `
<div>
  <div
    class="sc-pNWdM gAgbzS"
  >
    <div
      class="sc-gKAaRy ebxjGi"
      data-testid="collection-preview"
    >
      <a
        class="sc-iCoGMd bgiQSS"
        href="#/shop/jackets"
      >
        JACKETS
      </a>
      <div
        class="sc-fujyAs hAIBqS"
      >
        <div
          class="sc-gtsrHT Mxfvc"
          data-testid="collection-item"
        >
          <div
            class="sc-hKFxyN diAvRO"
            role="img"
          />
          <div
            class="sc-eCApnc flhIsE"
          >
            <span
              class="sc-jSFjdj duvpOf"
            >
              Jacket 1
            </span>
            <span>
              $10
            </span>
          </div>
          <button
            class="sc-bdnxRM pMdtA sc-dlnjwi kdkiUs"
          >
            Add to cart
          </button>
        </div>
        <div
          class="sc-gtsrHT Mxfvc"
          data-testid="collection-item"
        >
          <div
            class="sc-hKFxyN hCQygF"
            role="img"
          />
          <div
            class="sc-eCApnc flhIsE"
          >
            <span
              class="sc-jSFjdj duvpOf"
            >
              Jacket 2
            </span>
            <span>
              $20
            </span>
          </div>
          <button
            class="sc-bdnxRM pMdtA sc-dlnjwi kdkiUs"
          >
            Add to cart
          </button>
        </div>
      </div>
    </div>
    <div
      class="sc-gKAaRy ebxjGi"
      data-testid="collection-preview"
    >
      <a
        class="sc-iCoGMd bgiQSS"
        href="#/shop/shoes"
      >
        SHOES
      </a>
      <div
        class="sc-fujyAs hAIBqS"
      >
        <div
          class="sc-gtsrHT Mxfvc"
          data-testid="collection-item"
        >
          <div
            class="sc-hKFxyN hhfBmK"
            role="img"
          />
          <div
            class="sc-eCApnc flhIsE"
          >
            <span
              class="sc-jSFjdj duvpOf"
            >
              Shoes 1
            </span>
            <span>
              $100
            </span>
          </div>
          <button
            class="sc-bdnxRM pMdtA sc-dlnjwi kdkiUs"
          >
            Add to cart
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
